src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -1041,11 +1126,15 @@ void ov::npuw::JustInferRequest::unsafe_infer(std::size_t real_idx) {
     }
 }
 
-void ov::npuw::JustInferRequest::unsafe_run_this_prep_next(std::size_t idx, bool& next_prepared) {
+void ov::npuw::JustInferRequest::unsafe_run_this_prep_next(std::size_t idx, bool& next_prepared, bool& failover) {
```

Comment:
note - this method is called `unsafe_run_this_prep_next`. It didn't have any failover functionality before so it shouldn't have it now.

Reply:
I renamed `failover` to `accuracy_failover` so it will stay `unsafe` in terms of usual `failover`. I need `accuracy_failover` at this level as `unsafe_infer()` call `r->infer()` multiple times for spatial subgraph, so it is easer to wrap just `r->infer()` call with accuracy logic.

Reply:
let's decouple those.

accuracy check must happen after the inference is complete.

I know it is more convenient for you to plugin right into the `unsafe_infer()` but this code shouldn't be there.

Instead, where normally the accuracy check is happens, you need to run your reference subrequest over the same range.

You can borrow the range-based execution logic and generalize it so both `unsafe_infer` and the accuracy check use the same flow - the accuracy check shouldn't be injected that deep into it.